drizzlepac/wcs_functions.py

```diff
@@ -1069,15 +1070,17 @@ def make_mosaic_wcs(filenames, rot=None, scale=None):
     mosaic_wcs : HSTWCS object
         the merged composite WCS
     """
+    
+    # makes str into list of str
     if not isinstance(filenames, list):
         filenames = [filenames]
 
-    # Compile list of WCSs for all chips from all input filenames
-    hstwcs_list = []
-    for f in filenames:
-        hstwcs_list.extend([get_hstwcs(f, extnum) for extnum in get_extns(f)])
-    # Combine them into a single mosaic WCS
-    output_wcs = utils.output_wcs(hstwcs_list, undistort=True)
+    hstwcs_list = [get_hstwcs(f, extnum) for f in filenames for extnum in get_extns(f)]
+
+    # Generate output WCS based on input WCSs, and undistorts depending on extension.
```

Comment:
I'm not sure about the meaning of `undistorts depending on extension`. Could you clarify the comment?

Reply:
I will remove that part. 

The aim was to allow this to work for DRZ(C)s and FLT(C)s, but it turns out that much more code would have to change to make that a reality. For the moment, we are just using this as an opportunity to clean up this part of the code.